site/_posts/2020-05-27-CSI-integration.md

```diff
@@ -0,0 +1,201 @@
+- [`daemonset.apps/csi-azuredisk-node`](https://github.com/kubernetes-sigs/azuredisk-csi-driver/blob/master/deploy/csi-azuredisk-node.yaml)
+- [`deployment.apps/csi-snapshot-controller`](https://github.com/kubernetes-sigs/azuredisk-csi-driver/blob/master/deploy/csi-snapshot-controller.yaml)
+
+## Install Velero with CSI support enabled
+
+The CSI volume snapshot capability is currently, as of version 1.4, a beta feature behind the `EnableCSI` feature flag and is not enabled by default.
+
+Following instructions from our [docs website](https://velero.io/docs/csi/), install Velero with the [velero-plugin-for-csi](https://github.com/vmware-tanzu/velero-plugin-for-csi) and using the Azure Blob Store as our BackupStorageLocation. Please refer to our [velero-plugin-for-microsoft-azure documentation](https://github.com/vmware-tanzu/velero-plugin-for-microsoft-azure) for instructions on setting up the BackupStorageLocation. Please note that the BackupStorageLocation should be set up before installing Velero.
+
+Install Velero by running the below command
+
+```bash
+velero install \ 
+--provider azure \ 
+--plugins velero/velero-plugin-for-microsoft-azure:v1.1.0,velero/velero-plugin-for-csi:v0.1.1 \ 
+--bucket $BLOB_CONTAINER \ 
+--secret-file <PATH_TO_CREDS_FILE>/aks-creds \ 
+--backup-location-config resourceGroup=$AZURE_BACKUP_RESOURCE_GROUP,storageAccount=$AZURE_STORAGE_ACCOUNT_ID,subscriptionId=$AZURE_BACKUP_SUBSCRIPTION_ID \ 
+--snapshot-location-config apiTimeout=5m,resourceGroup=$AZURE_BACKUP_RESOURCE_GROUP,subscriptionId=$AZURE_BACKUP_SUBSCRIPTION_ID \ 
+--image velero/velero:v1.4.0 \ 
+--features=EnableCSI 
+```
+
+NOTE: Please use the latest released version of the container images from the [official Velero dockerhub](https://hub.docker.com/u/velero).
+
+## Deploy a CSI application
+
+Before installing the CSI application, install the storage class and the volume snapshot class for the Azure disk CSI driver by applying the below `yaml` to our cluster.
+
+```yaml
+apiVersion: storage.k8s.io/v1 
+kind: StorageClass 
+metadata: 
+  name: disk.csi.azure.com 
+provisioner: disk.csi.azure.com 
+parameters: 
+  skuname: StandardSSD_LRS 
+reclaimPolicy: Delete 
+volumeBindingMode: Immediate 
+allowVolumeExpansion: true
+---
+apiVersion: snapshot.storage.k8s.io/v1beta1 
+kind: VolumeSnapshotClass 
+metadata: 
+  name: csi-azuredisk-vsc 
+driver: disk.csi.azure.com 
+deletionPolicy: Retain 
+parameters:
+  tags: 'foo=aaa,bar=bbb' 
+```
+
+NOTE: The above `yaml` was sourced from [StorageClass](https://github.com/kubernetes-sigs/azuredisk-csi-driver/blob/master/deploy/example/storageclass-azuredisk-csi.yaml) and [VolumeSnapshotClass](https://github.com/kubernetes-sigs/azuredisk-csi-driver/blob/master/deploy/example/snapshot/storageclass-azuredisk-snapshot.yaml).
+
+
+Deploy a CSI application: a pod using a CSI backed PVC, in the `csi-app` namespace.
```

Comment:
Similar to what Jonas has said above, I would reword this to say something like "CSI-backed" application, not "CSI application"